test/modules/encoders/test_cnn_encoder.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import unittest
+
+import torch
+from torch import nn, Tensor
+from test.test_utils import assert_expected
+from torchmultimodal.modules.encoders.cnn_encoder import CNNEncoder
+class TestCnnEncoder(unittest.TestCase):
+    def test_invalid_arg_lengths(self):
+        input_dims = [1,2]
+        output_dims = [3,4,5]
+        kernel_sizes = [6,7,8]
+        self.assertRaises(
+            AssertionError, CNNEncoder, input_dims, output_dims, kernel_sizes
+        )
+
+    def test_invalid_output_dims(self):
+        input_dims = [0,1,2,3]
+        output_dims = [1,2,4,5]
+        kernel_sizes = [8,9,10,11]
+        self.assertRaises(
+            AssertionError, CNNEncoder, input_dims, output_dims, kernel_sizes
+        )
+
+    def test_single_layer_output_shape(self):
+        input = torch.zeros(5,3,128,128)
+        cnn_encoder = CNNEncoder([3],[3],[5])
+        actual = cnn_encoder(input)
+        expected = torch.zeros(5,12288)
+        assert_expected(actual, expected)
+
+    def test_multilayers_output_shape(self):
+        input = torch.zeros(5,3,128,128)
+        cnn_encoder = CNNEncoder([3,3,3],[3,3,3],[5,5,5])
+        zero_bias = nn.Parameter(Tensor([0.,0.,0.]))
+        for i in range(3):
+            cnn_encoder.cnn[i][0].bias = zero_bias
+        actual = cnn_encoder(input)
+        expected = torch.zeros(5,768)
+        assert_expected(actual, expected)
+
+    def test_varying_dims_and_kernels(self):
+        input = torch.rand(5,3,128,128)
+        cnn_encoder = CNNEncoder([3,2,1],[2,1,2],[3,5,7])
+        out = cnn_encoder(input)
+        self.assertEqual(out.size(), torch.Size([5,512]))
+
+    def test_fixed_weight_and_bias(self):
+        input = Tensor(
+            [
+                [[[1,2,3],[4,5,6]]],
```

Comment:
Similar comment for unsqueeze here. Also, since you are using this tensor in multiple tests you can assign it as an instance variable in the `setUp` method. Then the tensor will only get constructed once when running the entire test module.